File: hLJDecision/src/main/java/com/hlj/fragment/ContactUsFragment.java
package com.hlj.fragment;

import shawn.cxwl.com.hlj.R;
import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.hlj.activity.HPDFActivity;
import com.hlj.common.CONST;

/**
 * 联系我们
 * @author shawn_sun
 *
 */

public class ContactUsFragment extends Fragment{
	
	private TextView tvData,tvPhone;

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_contact_us, null);
		return view;
	}
	
	@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		tvData = (TextView) view.findViewById(R.id.tvData);
		tvPhone = (TextView) view.findViewById(R.id.tvPhone);

		tvData.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View v) {
				Intent intent = new Intent(getActivity(), HPDFActivity.class);
				intent.putExtra(CONST.ACTIVITY_NAME, "部分产品数据说明");
				intent.putExtra(CONST.WEB_URL, "http://decision-admin.tianqi.cn/data/%E9%BB%91%E9%BE%99%E6%B1%9F%E6%B0%94%E8%B1%A1%E6%89%8B%E6%9C%BAAPP%E9%83%A8%E5%88%86%E4%BA%A7%E5%93%81%E6%95%B0%E6%8D%AE%E8%AF%B4%E6%98%8E.pdf");
				startActivity(intent);
			}
		});

		tvPhone.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View arg0) {
//				dialPhone("联系电话", "0451-55172953", "拨打");
			}
		});
	}
	
	private void dialPhone(String message, final String content, String positive) {
		LayoutInflater inflater = (LayoutInflater) getActivity().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
		View view = inflater.inflate(R.layout.delete_dialog, null);
		TextView tvMessage = (TextView) view.findViewById(R.id.tvMessage);
		TextView tvContent = (TextView) view.findViewById(R.id.tvContent);
		LinearLayout llNegative = (LinearLayout) view.findViewById(R.id.llNegative);
		LinearLayout llPositive = (LinearLayout) view.findViewById(R.id.llPositive);
		TextView tvPositive = (TextView) view.findViewById(R.id.tvPositive);
		
		final Dialog dialog = new Dialog(getActivity(), R.style.CustomProgressDialog);
		dialog.setContentView(view);
		dialog.show();
		
		tvPositive.setText(positive);
		tvMessage.setText(message);
		tvContent.setText(content);
		tvContent.setVisibility(View.VISIBLE);
		llNegative.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View arg0) {
				dialog.dismiss();
			}
		});
		
		llPositive.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View arg0) {
				dialog.dismiss();
				startActivity(new Intent(Intent.ACTION_CALL, Uri.parse("tel:"+content)));
			}
		});
	}
	
}

File: hLJDecision/src/main/java/com/hlj/fragment/ContactUsFragment.kt
package com.hlj.fragment

import android.app.Dialog
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.hlj.activity.DataIntroActivity
import com.hlj.common.CONST
import com.hlj.common.MyApplication
import com.hlj.utils.CommonUtil
import kotlinx.android.synthetic.main.dialog_delete.view.*
import kotlinx.android.synthetic.main.fragment_contact_us.*
import shawn.cxwl.com.hlj.R

/**
 * 联系我们
 * @author shawn_sun
 */
class ContactUsFragment : Fragment() {

    private var mReceiver: MyBroadCastReceiver? = null
    private var desc = ""

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_contact_us, null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setDesc()
        initBroadCast()
    }

    private fun setDesc() {
        for (i in 0 until MyApplication.columnDataList.size) {
            val dto1 = MyApplication.columnDataList[i]
            var desc1 = "${(i+1)}.${dto1.name}：${dto1.desc}\n"
            for (j in 0 until dto1.child.size) {
                val dto2 = dto1.child[j]
                val desc2 = "   ${(i+1)}.${(j+1)}.${dto2.name}：${dto2.desc}\n"
                desc1 += desc2
            }
            desc+=desc1
        }
    }

    private fun initBroadCast() {
        mReceiver = MyBroadCastReceiver()
        val intentFilter = IntentFilter()
        intentFilter.addAction(arguments!!.getString(CONST.BROADCAST_ACTION))
        activity!!.registerReceiver(mReceiver, intentFilter)
    }

    private inner class MyBroadCastReceiver : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            refresh()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        if (mReceiver != null) {
            activity!!.unregisterReceiver(mReceiver)
        }
    }

    private fun refresh() {
        ivData!!.setOnClickListener {
            val intent = Intent(activity, DataIntroActivity::class.java)
            intent.putExtra(CONST.ACTIVITY_NAME, "客户端产品数据说明")
            intent.putExtra("desc", desc)
            startActivity(intent)
        }
//        tvPhone.setOnClickListener {
//            dialPhone("联系电话", "0451-55172953", "拨打");
//        }
        val columnId = arguments!!.getString(CONST.COLUMN_ID)
        val title = arguments!!.getString(CONST.ACTIVITY_NAME)
        CommonUtil.submitClickCount(columnId, title)
    }

    private fun dialPhone(message: String, content: String, positive: String) {
        val inflater = activity!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = inflater.inflate(R.layout.dialog_delete, null)
        val dialog = Dialog(activity, R.style.CustomProgressDialog)
        dialog.setContentView(view)
        dialog.show()
        view.tvPositive.text = positive
        view.tvMessage.text = message
        view.tvContent.text = content
        view.tvContent.visibility = View.VISIBLE
        view.llNegative.setOnClickListener { dialog.dismiss() }
        view.llPositive.setOnClickListener {
            dialog.dismiss()
            startActivity(Intent(Intent.ACTION_CALL, Uri.parse("tel:$content")))
        }
    }
	
}
